packages/block-library/src/navigation/index.php

```diff
@@ -67,6 +67,41 @@ function build_css_font_sizes( $attributes ) {
 	return $font_sizes;
 }
 
+/**
+ * Filters out links with no labels.
+ *
+ * @param array $block Block to check label attribute on.
+ *
+ * @return boolean
+ */
+function gutenberg_has_navigation_label( $block ) {
+	return ! empty( $block['attrs']['label'] );
+}
+
+/**
+ * Recursively filters out links with no labels to build a clean navigation block structure.
+ *
+ * @param array $blocks Navigation link inner blocks from the Navigation block.
+ *
+ * @return boolean
+ */
+function gutenberg_remove_empty_navigation_links_recursive( $blocks ) {
+
+	$blocks = array_filter( $blocks, 'gutenberg_has_navigation_label' );
+
+	if ( empty( $blocks ) ) {
+		return $blocks;
+	}
+
+	foreach ( $blocks as $key => $val ) {
```

Comment:
Personal preference: `$block` would be more descriptive than `$val` and it could be used in the following `empty()` check and passed to the function call instead of `$blocks[ $key ]` to make it a bit easier to read.